File: API/Features/Fighters/Models/Fighter.cs
using API.Features.Brackets.Models;
using API.Features.Clubs.Models;
using API.Features.Competitions.Models;

namespace API.Features.Fighters.Models;

public class Fighter
{
    public DateTime Birthdate { get; set; }
    public ICollection<Bracket> Brackets { get; set; } = [];
    public Club? Club { get; set; }
    public string? ClubId { get; set; }
    public ICollection<Competition> Competitions { get; set; } = [];
    public string? Gender { get; set; }
    public string? Id { get; set; }
    public string? Name { get; set; }
    public string? Rank { get; set; }
    public float Weight { get; set; }
}

File: Client/src/app/fighters/fighter.ts
import { ClubDto } from '../clubs/club';
import { Page } from '../shared/models/page';

export interface CreateFighterDto {
  name: string;
  gender: string;
  birthDate: string;
  weight: number;
  clubId: string;
  rank: string;
}

export interface FighterDto {
  id: string;
  name: string;
  gender: string;
  birthdate: string;
  weight: number;
  club: string;
  rank: string;
  competitionIds: string[];
  matchIds: string[];
}

export interface ShiftPage extends Page<FighterDto> {}
